### PR TITLE
Remote API

### PR DESCRIPTION
**Pull request by danimoh**
## Pull request checklist

- [X] All tests pass. Demo project builds and runs.
- [X] I have resolved any merge conflicts. Confirmation: ____

## What's in this pull request?

Remote API

### CODE REVIEW

File: clients/remote-core/server/RemoteBlockchainAPI.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        } catch(e) {
+            connection.sendError('A valid block hash in Base64 format required.', RemoteBlockchainAPI.Commands.BLOCKCHAIN_GET_BLOCK);
+            return;
+        }
+        this.$.blockchain.getBlock(hash)
+            .then(block => connection.send(RemoteBlockchainAPI.MessageTypes.BLOCKCHAIN_BLOCK, {
+                block: this._serializeToBase64(block),
+                hash: hashString
+            }))
+            .catch(e => connection.sendError('Failed to get block '+hashString+' - '+e, RemoteBlockchainAPI.Commands.BLOCKCHAIN_GET_BLOCK));
```
Comment (paberr) [inline clients/remote-core/server/RemoteBlockchainAPI.js]: For the first string, you could use backticks, i.e., `` `Failed to get block ${hashString} - ${e}` ``

File: clients/remote-core/server/RemoteBlockchainAPI.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+    /**
+     * @private
+     * Send the next compact target of the blockchain.
+     * @param {AuthenticatedConnection} connection - The requesting connection.
+     */
+    _sendNextCompactTarget(connection) {
+        this.$.blockchain.getNextCompactTarget()
+            .then(nextCompactTarget => connection.send(RemoteBlockchainAPI.MessageTypes.BLOCKCHAIN_NEXT_COMPACT_TARGET, nextCompactTarget))
+            .catch(e => connection.sendError('Failed to get next compact target.', RemoteBlockchainAPI.Commands.BLOCKCHAIN_GET_NEXT_COMPACT_TARGET));
```
Comment (paberr) [inline clients/remote-core/server/RemoteBlockchainAPI.js]: This error ends with a dot, while others don't.

File: clients/remote-core/client/RemoteMempool.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+                }
+                this.fire(RemoteMempool.Events.TRANSACTIONS_READY);
+            });
+        } else {
+            super._handleEvents(message);
+        }
+    }
+}
+RemoteMempool.IDENTIFIER = 'mempool';
+RemoteMempool.ATTRIBUTES = [];
```
Comment (paberr) [inline clients/remote-core/client/RemoteMempool.js]: It is a bit unintuitive here that no attributes are given although the `transactions` attribute is transferred.
I would at least put a comment here.

File: clients/remote-core/client/RemoteClass.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+            }
+            return state;
+        });
+    }
+
+    _handleEvents(message) {
+        const serverEvent = message.type;
+        const clientEvent = this._removeNameSpace(serverEvent);
+        if (this._events.indexOf(clientEvent)!==-1) {
+            this.fire(clientEvent, message.data);
```
Comment (danimoh) [inline clients/remote-core/client/RemoteClass.js]: Yep, thats true. Are there events that fire with multiple arguments ?

Comment (paberr) [inline clients/remote-core/client/RemoteClass.js]: Yes, there are: https://github.com/nimiq-network/core/search?utf8=✓&q=.fire&type=

**Pull request closed without merging**